hudi-sync/hudi-sync-common/src/main/java/org/apache/hudi/sync/common/HoodieSyncClient.java

```diff
@@ -83,18 +87,24 @@ public boolean isBootstrap() {
     return metaClient.getTableConfig().getBootstrapBasePath().isPresent();
   }
 
-  public boolean isDropPartition() {
+  /**
+   * Get the set of dropped partitions based on the latest commit metadata.
+   * Returns empty set if the latest commit was not due to DELETE_PARTITION operation.
+   */
+  public Set<String> getDroppedPartitions() {
     try {
-      Option<HoodieCommitMetadata> hoodieCommitMetadata = HoodieTableMetadataUtil.getLatestCommitMetadata(metaClient);
+      Option<HoodieCommitMetadata> hoodieCommitMetadata = getLatestCommitMetadata(metaClient);
 
       if (hoodieCommitMetadata.isPresent()
           && WriteOperationType.DELETE_PARTITION.equals(hoodieCommitMetadata.get().getOperationType())) {
-        return true;
+        Map<String, List<String>> partitionToReplaceFileIds =
+            ((HoodieReplaceCommitMetadata) hoodieCommitMetadata.get()).getPartitionToReplaceFileIds();
```

Comment:
Why are we doing getpartitionToReplaceFileIds() here? `DELETE_PARTITION` operation can occur in a normal commit other than replacecommit action or am I missing anything here?

Reply:
DELETE_PARTITION operation adds a replacecommit action in the timeline. The partitions which get deleted are contained in `partitionToReplaceFileIds `.

Reply:
I see.